clowder-framework/clowder
# Ability to queue archive / unarchive for full datasets

## Description


### Problem
The new auto-archiving feature is very nice, but adds the possibility that many files could be accidentally archived en masse. Unarchiving, in this case, would be very tedious as each file needs to be unarchived individually, which requires a lot of UI navigation, clicking, and waiting.

### Approach
Save the user in this case by offering an "Archive All" and "Unarchive All" button at the dataset-level

These buttons will quickly recurse over the contents of the dataset and queue each file inside for archival / unarchival respectively.

I've also added a new Permission for `ArchiveDataset` (since we previously added `ArchiveFile`, we should match that pattern)

### How to Test
Prerequisite: Minio running, Clowder configured to use `S3ByteStorageDriver`, S3 archival extractor configured and running

1. Checkout and run this branch locally
2. Login to Clowder
3. Create a dataset, if you haven't already
4. Within the dataset, upload a file at the root level and create a folder
5. Navigate into the folder you just created, upload a file, and create another folder
6. Navigate once again into this folder and upload one last file
    * You should now have a dataset that looks roughly as follows:
        * Dataset
            * Folder A
                * Folder B
                   * File 3
                * File 2
            * File 1

## Review Time Estimate


## Types of changes


## Checklist:
